client/src/Checkout.elm

```diff
@@ -530,6 +631,80 @@ update msg model authStatus maybeSessionToken checkoutDetails =
         RefreshDetails _ ->
             model |> nothingAndNoCommand
 
+        HelcimCheckoutTokenReceived (RemoteData.Success (Ok token)) ->
+            ( { model | checkoutToken = Just token.checkoutToken }, Nothing
+            , Cmd.batch [ Ports.appendHelcimPayIframe token.checkoutToken, Ports.subscribeToHelcimMessages () ]
+            )
+
+        HelcimCheckoutTokenReceived (RemoteData.Success (Err errors)) ->
+            { model | errors = errors, isPaymentProcessing = False } |> nothingAndNoCommand
+
+        HelcimCheckoutTokenReceived (RemoteData.Failure error) ->
+            { model | errors = Api.apiFailureToError error, isPaymentProcessing = False } |> nothingAndNoCommand
+
+        HelcimCheckoutTokenReceived _ ->
+            { model | isPaymentProcessing = False } |> nothingAndNoCommand
+
+        HelcimEventReceived event -> case Decode.decodeValue helcimPayEventDecoder event of
+            Err _ ->
+                { model | errors = Api.addError "" ("Unexpected HelcimPay.js response.") model.errors, isPaymentProcessing = False } |> nothingAndNoCommand
+
+            Ok (HelcimPayInit _) -> model |> nothingAndNoCommand
+
+            Ok (HelcimPayCheckout checkoutEvent) ->
+                case checkoutDetails of
```

Comment:
This logic is kinda complicated and I would prefer to move it to a separate function.